app/src/main/java/com/codesoom/assignment/application/AuthenticationService.java

```diff
@@ -18,19 +19,40 @@ public AuthenticationService(UserRepository userRepository,
         this.jwtUtil = jwtUtil;
     }
 
+    /**
+     * 회원 로그인을 처리하고 토큰을 반환합니다.
+     * @param email 로그인할 이메일
+     * @param password 비밀번호
+     * @return 토큰
+     * @throws LoginFailException 회원을 찾지 못하거나 비밀번호가 틀린 경우
+     */
     public String login(String email, String password) {
-        User user = userRepository.findByEmail(email)
-                .orElseThrow(() -> new LoginFailException(email));
+        return userRepository.findByEmail(email)
+                .map(user -> {
+                    if (!user.authenticate(password)) {
+                        throw new LoginFailException(email);
```

Comment:
```suggestion
 return null;
```

이렇게 하면 예외를 던지는 부분이 한곳으로 일관되지않을까 싶습니다! 동일한 예외를 던지는 것이 목적이라면요!

Reply:
넵 동일한 예외를 던지도록 하는거였는데 수정된 코드가 일관적이고 좋은 것 같아요! 반영했습니다